### DISCUSSION TITLE
Panels - dismiss panel functions don't wait for animation to be complete

### DISCUSSION THREAD

**Discussion by shaz01** [arkivanov/Decompose#809] [Q&A]
nav.dismissDetails(
			onComplete = { _, _ ->
				mainScope.launch {
					delay(75) // this is needed, but shouldn't be
					nav.setMode(SINGLE)
				}
			}
		) 
		
		
this code shows the details panel for a split second as the single view, when it really should be showing main panel. (mainScope.launch is for delay only and isn't the cause)

**Comment by arkivanov** [comment]
This is by design. The core navigation (in the main `decompose` module) is not aware of the UI. There are a few options.

1. You can try showing a placeholder component instead of dismissing the details component.
2. You really want to dismiss the details component and switch to SINGLE mode, you can switching the mode from the UI. E.g. with Compose, you can try switching the mode from `DisposableEffect { onDispose {...} }` callback of the details composable function.
3. You can try implementing your own `ChildPanelsLayout` with proper animations between modes, but this might be not so easy.
